Orange/widgets/data/owdatasets.py

```diff
@@ -456,20 +457,24 @@ def set_model(self, model, current_index):
                 QItemSelectionModel.ClearAndSelect | QItemSelectionModel.Rows)
             self.commit()
 
+    def indicator_state_for_info(self, info):
+        localinfo = list_local(self.local_cache_path)
```

Comment:
This is called in a for loop so it reads all info files from disk and parses the json info for each dataset.
Quadratic complexity for cpu operations and in-ram data would not be a problem, but we should probably avoid excessive disk access.

Reply:
This function is used in two places, both time in a loop. I change it to require `localinfo` as an argument, which can be computed before the loop.

Someday, we should refactored. Instead of this loop, we should use a widget-specific model instead of `QStandardItemModel`.